components/src/dynamo/vllm/handlers.py

```
@@ -1319,6 +1339,59 @@ async def sleep(self, body: dict) -> dict:
+        """Deepen an existing sleep without rejoining discovery.
+
+        vLLM cannot deepen a sleep in place, so the engine is briefly woken
+        and re-slept at the deeper level. Generation stays paused and the
+        endpoint stays unregistered throughout, so no requests can arrive.
+        """
+        async with self._pause_lock:
+            if not self._pause_controller.is_paused:
+                return {"status": "error", "message": "Engine is not sleeping"}
+            try:
+                await self.engine_client.wake_up()
+            except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)